src/frontend/discussions-abusefilter-app/src/interfaces/BackendInterface.ts

```diff
@@ -0,0 +1,19 @@
+import type { Filter, FilterMetadata } from '@shared/filters';
+
+export type Error = {
+  error: {
+    message: string
+  }
+};
+
+export const interfaceMode = 'local';
+
+export interface BackendInterface {
+  getFilters: () => Promise<FilterMetadata[]>,
+  getFilter: (filterId: string) => Promise<{ filterDetails: Filter, filterMetadata: FilterMetadata }> | Error,
+  createFilter: (filter: Filter, filterMetadata: FilterMetadata) => void,
+  updateFilter: (filter: Filter, filterMetadata: FilterMetadata) => void,
+  archiveFilter: (filterId: string) => void,
+  enableFilter: (filterId: string) => void,
+  disableFilter: (filterId: string) => void,
```

Comment:
These should all ideally return promises (`=> Promise<void>`), though it's fine to keep it as this for now. For example, we'd want to know if enabling a filter is successful or not